sshd-core/src/main/java/org/apache/sshd/agent/common/AbstractAgentProxy.java

```diff
@@ -71,10 +82,17 @@ public void setShutdownOnExit(boolean shutdown) {
 
     @Override
     public List<Pair<PublicKey, String>> getIdentities() throws IOException {
-        Buffer buffer = createBuffer(SshAgentConstants.SSH2_AGENTC_REQUEST_IDENTITIES, 1);
+        byte cmd = SshAgentConstants.SSH2_AGENTC_REQUEST_IDENTITIES;
+        byte okcmd = SshAgentConstants.SSH2_AGENT_IDENTITIES_ANSWER;
+        if (FactoryManager.AGENT_FORWARDING_TYPE_IETF.equals(channelType)) {
+            cmd = SshAgentConstants.SSH_AGENT_LIST_KEYS;
+            okcmd = SshAgentConstants.SSH_AGENT_KEY_LIST;
+        }
+
+        Buffer buffer = createBuffer(cmd, 1);
         buffer = request(prepare(buffer));
         int type = buffer.getUByte();
-        if (type != SshAgentConstants.SSH2_AGENT_IDENTITIES_ANSWER) {
+        if (type != okcmd) {
```

Comment:
This does not seem equivalent to previous code - you are **creating** a new buffer whereas the previous code was not doing that.